pgx-macros/src/lib.rs

```diff
@@ -64,19 +64,41 @@ pub fn pg_test(attr: TokenStream, item: TokenStream) -> TokenStream {
         }
     });
 
-    stream.extend(proc_macro2::TokenStream::from(pg_extern(
-        attr,
-        item.clone(),
-    )));
-
-    let expected_error = match expected_error {
-        Some(msg) => quote! {Some(#msg)},
-        None => quote! {None},
-    };
-
     let ast = parse_macro_input!(item as syn::Item);
+
     match ast {
-        Item::Fn(func) => {
+        Item::Fn(mut func) => {
+            // Here we need to break out attributes into test and non-test attributes,
+            // so the generated #[test] attributes are in the appropriate place.
+            let mut test_attributes = Vec::new();
+            let mut non_test_attributes = Vec::new();
+
+            for attribute in func.attrs.iter() {
+                if let Some(ident) = attribute.path.get_ident() {
+                    let ident_str = ident.to_string();
+
+                    if ident_str == "ignore" || ident_str == "should_panic" {
+                        test_attributes.push(attribute.clone());
+                    } else {
+                        non_test_attributes.push(attribute.clone());
+                    }
```

Comment:
Remind me in about 8 weeks to show you how to fix this. There's another way to write this that doesn't "stutter" like this today, but this kind of stuttering is solved much more nicely by an upcoming syntactic nicety.

Reply:
@workingjubilee Looking forward to it :)